src/core/ddsi/src/ddsi_cdrstream.c

```diff
@@ -434,6 +434,9 @@ static uint32_t get_collection_elem_size (uint32_t insn, const uint32_t * __rest
     case DDS_OP_VAL_EXT:
       abort ();
       break;
+    default:
+      // unexpected value
+      abort ();
```

Comment:
Stealth fix of warning to appease ROS 2 but also to avoid returning an uninitialized value in the default case (can this happen?). We could also initialize the value with zero upon declaration but this a) is likley not optimized away in all other cases and b) returns still a zero in all non-handled cases (default, is this logically acceptable?)